### PR TITLE
fix: catch download errors

### PR DESCRIPTION
**Pull request by hal-shu-sato**
## Description

- Fix to catch download errors.

## Motivation and Context

When a file to be downloaded doesn't exist, apm stops working.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -493,24 +493,29 @@ function launch() {
...
+            path.dirname(savePath),
+            getHash(keyText) + '_' + path.basename(savePath)
+          );
+          fs.renameSync(savePath, renamedPath);
+          savePath = renamedPath;
+        }
+        return savePath;
+      } catch (e) {
+        log.error(e);
+        return false;
```
Comment (mitosagi) [inline src/main.ts:517]: It can simplify function types.
```suggestion
        return undefined;
```

**Review by mitosagi — COMMENTED** [review]
It' s a nice change that will make debugging easier.

File: src/lib/convertId.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ async function getIdDict(update = false): Promise<{ [key: string]: string }> {
       'package',
       dictUrl
     );
-    return fs.readJsonSync(convertJson);
+    return dictUrl ? fs.readJsonSync(convertJson) : {};
```
Comment (mitosagi) [inline src/lib/convertId.ts:23]: Do you mean convertJson?
```suggestion
    return convertJson ? fs.readJsonSync(convertJson) : {};
```

**Comment by hal-shu-sato** [comment]
I fixed them.

**Review by mitosagi — APPROVED** [review]

**Pull request merged**